### ISSUE TITLE
New TMB dependency on `make` in Windows - reducing reliability

### ISSUE THREAD

**Issue by arni-magnusson** [kaskr/adcomp#289]
It looks like TMB 1.7.14 introduced a dependency on `make` in Windows that was not the case in 1.7.13 and earlier versions of TMB.

In other words, one used to be able to build and run TMB models in Windows without having `make` in the PATH. In version 1.7.14 `make` needs to be in the PATH to build a TMB model.

The reason why the dependency on `make` in Windows is important is because many machines will have PATH entries that now prevent TMB from working. When a user installs Rtools and TMB, there may be earlier PATH entries that prevent the user from building TMB models. See:

https://github.com/kaskr/adcomp/issues/284#issuecomment-428627978

As far as I can tell, the error message `fatal error: R_ext/Print.h: No such file or directory #include <R_ext/Print.h>` is likely to pop up on Windows computers that have multiple copies of `sh.exe` in the PATH. An error that did not occur in versions 1.7.13 and earlier.

***
*Reproducible example*

Build a TMB model on a Windows machine with TMB 1.7.14 that has `sh.exe` from any software other than Rtools earlier in the PATH. [Crashes with the `R_ext/Print.h` error message]

***

In short, after users installed Rtools and TMB in the past, they used to be quite guaranteed to have a working TMB environment, but now it depends more on what's already installed on the computer. There are many good reasons why users may have more than one `sh.exe` in their PATH. Something seems to have changed between 1.7.13 and 1.7.14 that introduced the `make` dependency in Windows, but I haven't been able to identify what changed in the source code to introduce the dependency.

If the benefits of introducing the `make` dependency in Windows outweigh the increased likelihood of user errors, then all is good. I just wanted to highlight this issue, as I haven't seen anything about it in the NEWS file or other documentation.

**Comment by kaskr** [comment]
> It looks like TMB 1.7.14 introduced a dependency on make in Windows

There are no such changes in TMB: `git diff v1.7.13 v1.7.14`
Sounds more like an Rtools issue ?

**Comment by arni-magnusson** [comment]
2018 discussion, probably not relevant in 2023.
